MdeModulePkg/Bus/Pci/XhciDxe/XhciSched.c

```diff
@@ -1713,6 +1719,10 @@ XhcMonitorAsyncRequests (
       CopyMem (ProcBuf, Urb->Data, Urb->Completed);
     }
 
+    // The update call must occur before the callback since the callback
+    // may remove and free the URB, leading to a fault.
+    XhcUpdateAsyncRequest (Xhc, Urb);

```

Comment:
`XhcUpdateAsyncRequest` updates `Urb` including some of the fields passed to the callback: https://github.com/tianocore/edk2/blob/bdca3681e90b924bfa89b378c2687eef4e22db91/MdeModulePkg/Bus/Pci/XhciDxe/XhciSched.c#L247. Is that okay? Do we need to save off these values and pass them to the callback?

Looking at an example callback, it looks like we would need to: https://github.com/tianocore/edk2/blob/bdca3681e90b924bfa89b378c2687eef4e22db91/MdeModulePkg/Bus/Usb/UsbMouseDxe/UsbMouse.c#L744.